app/api/chemotion/sample_api.rb

```diff
@@ -127,21 +110,22 @@ class SampleAPI < Grape::API
             tmp_file_path = File.join('tmp', temp_filename)
             # Write the contents of the uploaded file to the temporary file
             File.binwrite(tmp_file_path, file[:tempfile].read)
-            ImportSamplesJob.perform_later(
-              tmp_file_path,
-              params[:currentCollectionId],
-              current_user.id,
-              file['filename'],
-            )
+            parameters = {
+              collection_id: params[:currentCollectionId],
+              user_id: current_user.id,
+              file_name: file['filename'],
+              file_path: tmp_file_path,
+            }
+            ImportSamplesJob.perform_later(parameters)
             { status: 'in progress', message: 'Importing samples in background' }
           end
```

Comment:
What happens, if it is not Sdf, mol, csv or xls format. I think it will go into the xls/csv branch and crash somewhere there. Maybe a rescue needed ?

Edited: i saw that in the import_samples_job.rb there is a case statement which checks if format is .xlsx or .sdf and otherwise throws and error.